### PR TITLE
feat(migration): migrated code to public repo

### PR DESCRIPTION
**Pull request by cl-shaharshavit-rnd**

### CODE REVIEW

File: src/DAL/ImageDataRepository.ts [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+      this.mcLogger.info('attempt to update non existing record has occurred');
+      //TODO: replace with custom exception type and handle in service
+      throw new Error('invalid update');
+    }
+  }
+
+  public async deleteImageData(id: string): Promise<void> {
+    const image = await this.get(id);
+    if (image) {
+      await this.delete({ id: id });
```
Comment (asafMasa) [inline src/DAL/ImageDataRepository.ts:49]: add log before actualy deleting / updating or adding new image

File: src/DAL/ImageDataRepository.ts [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+    } else {
+      this.mcLogger.info('attempt to delete non existing record has occurred');
+      //TODO: replace with custom exception type and handle in service
+      throw new Error('image to delete was not found');
```
Comment (asafMasa) [inline src/DAL/ImageDataRepository.ts:53]: add the id to the error log

File: src/DAL/ImageDataRepository.ts [hunk 3]
```diff
@@ -0,0 +1,53 @@
...
+  }
+
+  public async updateImageDate(image: ImageData): Promise<void> {
+    const exists = (await this.get(image.id)) != null;
+    if (exists) {
+      await this.save(image);
+    } else {
+      this.mcLogger.info('attempt to update non existing record has occurred');
+      //TODO: replace with custom exception type and handle in service
+      throw new Error('invalid update');
```
Comment (asafMasa) [inline src/DAL/ImageDataRepository.ts:41]: add data to error log

File: src/DAL/ImageDataRepository.ts [hunk 4]
```diff
@@ -0,0 +1,53 @@
...
+  public async get(id: string): Promise<ImageData | undefined> {
+    return this.findOne({ id: id });
+  }
+
+  public async createAndSave(image: ImageData): Promise<void> {
+    const exists = (await this.get(image.id)) != undefined;
+    if (!exists) {
+      await this.save(image);
+    } else {
+      this.mcLogger.info('duplicate value error has occurred');
```
Comment (asafMasa) [inline src/DAL/ImageDataRepository.ts]: add data to error log

File: src/migration/1602141176877-migration-V1_0_0.ts [hunk 5]
```diff
@@ -0,0 +1,14 @@
+import {MigrationInterface, QueryRunner} from "typeorm";
```
Comment (asafMasa) [inline src/migration/1602141176877-migration-V1_0_0.ts:1]: if it's generated code why it's not on gitignore?

File: src/migration/migration.js [hunk 6]
```diff
@@ -0,0 +1,25 @@
...
+(err, stdout, stderr) => {
+    if (err) {
+      //some err occurred
+      console.error(err)
+    } else {
+     // the *entire* stdout and stderr (buffered)
+     console.log(`stdout: ${stdout}`);
+     console.log(`stderr: ${stderr}`);
+    }
+})
```
Comment (asafMasa) [inline src/migration/migration.js]: end of file

File: src/services/ImageService.ts [hunk 7]
```diff
@@ -0,0 +1,72 @@
...
+export class ImagesService {
+  private repository?: ImageDataRepository;
+  public constructor(private readonly connectionManager: ConnectionManager) {}
+
+  public async get(id: string): Promise<ImageMetadata> {
+    const repository = await this.getRepository();
+    const image = await repository.get(id);
+    if (!image) {
+      //TODO: replace with custom error type
+      throw new Error('image data was not found');
```
Comment (asafMasa) [inline src/services/ImageService.ts:17]: return data (id) on error log

**Review by asafMasa — CHANGES_REQUESTED** [review]

File: src/migration/1602141176877-migration-V1_0_0.ts (hunk 5 above)
Comment (cl-shaharshavit-rnd) [reply src/migration/1602141176877-migration-V1_0_0.ts:1]: it is required to update db between versions properly if it changes (and for rolling back db version).

File: src/DAL/ImageDataRepository.ts (hunk 1 above)
Comment (cl-shaharshavit-rnd) [reply src/DAL/ImageDataRepository.ts:49]: added

File: src/DAL/ImageDataRepository.ts (hunk 2 above)
Comment (cl-shaharshavit-rnd) [reply src/DAL/ImageDataRepository.ts:53]: error handling here is temporary will be handled in separate task/pr

File: src/DAL/ImageDataRepository.ts (hunk 3 above)
Comment (cl-shaharshavit-rnd) [reply src/DAL/ImageDataRepository.ts:41]: error handling here is temporary will be handled in separate task/pr

File: src/DAL/ImageDataRepository.ts (hunk 4 above)
Comment (cl-shaharshavit-rnd) [reply src/DAL/ImageDataRepository.ts]: error handling here is temporary will be handled in separate task/pr

File: src/migration/migration.js (hunk 6 above)
Comment (cl-shaharshavit-rnd) [reply src/migration/migration.js]: fixed

File: src/services/ImageService.ts (hunk 7 above)
Comment (cl-shaharshavit-rnd) [reply src/services/ImageService.ts:17]: error handling here is temporary will be handled in separate task/pr

File: src/DAL/ImageDataRepository.ts [hunk 8]
```diff
@@ -21,6 +21,7 @@ export class ImageDataRepository extends Repository<ImageData> {
   public async createAndSave(image: ImageData): Promise<void> {
     const exists = (await this.get(image.id)) != undefined;
     if (!exists) {
+      this.mcLogger.info(`creating ImageData record with id: ${image.id}`);
       await this.save(image);
     } else {
       this.mcLogger.info('duplicate value error has occurred');
```
Comment (asafMasa) [inline src/DAL/ImageDataRepository.ts]: this.mcLogger.info('duplicate value error has occurred'); - warn?
add id

File: readme.md [hunk 9]
```diff
@@ -1,41 +1,5 @@
...
-usage example:
-
-```typescript
-public constructor(
-    @inject(delay(() => MCLogger)) private readonly logger: MCLogger) {
-  }
-```
+### notes 
+please run ```npm run migration``` and commit changes before every tag (```npm run release```).
+this is required to keep db updates match the service version
```
Comment (asafMasa) [inline readme.md:5]: add description of the CRUD methods with example

File: src/DAL/ImageDataRepository.ts [hunk 10]
```diff
@@ -0,0 +1,56 @@
...
+    }
+  }
+
+  public async updateImageDate(image: ImageData): Promise<void> {
+    const exists = (await this.get(image.id)) != null;
+    if (exists) {
+      this.mcLogger.info(`updating ImageData record with id: ${image.id}`);
+      await this.save(image);
+    } else {
+      this.mcLogger.info('attempt to update non existing record has occurred');
```
Comment (asafMasa) [inline src/DAL/ImageDataRepository.ts]: warn

File: src/DAL/ImageDataRepository.ts [hunk 11]
```diff
@@ -0,0 +1,56 @@
...
+    }
+  }
+
+  public async deleteImageData(id: string): Promise<void> {
+    const image = await this.get(id);
+    if (image) {
+      this.mcLogger.info(`deleting ImageData record with id: ${image.id}`);
+      await this.delete({ id: id });
+    } else {
+      this.mcLogger.info('attempt to delete non existing record has occurred');
```
Comment (asafMasa) [inline src/DAL/ImageDataRepository.ts]: warn

Comment (asafMasa) [reply src/DAL/ImageDataRepository.ts]: and add the image.id

**Review by asafMasa — CHANGES_REQUESTED** [review]
Good Job! 🥇

**Review by asafMasa — APPROVED** [review]

**Pull request merged**